common/on_actions/babylon_infamy_rebalance_on_actions.txt

```diff
@@ -1,8 +1,50 @@
 ﻿on_babylon_monthly_pulse_country = {
-	#updates provinces each month
+	effect = {
+		set_local_variable = {
+			name = total_infrastructure
+			value = 0
+		}
+		set_local_variable = {
+			name = number_of_states
+			value = 0
+		}
+		every_scope_state = {
+			change_local_variable = {
+				name = total_infrastructure
+				add = modifier:state_infrastructure_add		
+			}
+			change_local_variable = {
+				name = number_of_states
+				add = 1
+			}
+		}
+
+		set_local_variable = {
+			name = gained_land_trade_capcity
+			value = {
+				add = local_var:total_infrastructure
+				divide = local_var:number_of_states
+				subtract = 50 # counting from 50 infrasturucture
+				divide = 8 # every 8 infrastructure average is 1 land_trade capacity
+			}
+		}
+
+		if = {
+			limit = {
+				local_var:gained_land_trade_capcity >= 1
+			}
+
+			add_modifier = { 
+				name = modifier_land_trade_capacity
+				multiplier = local_var:gained_land_trade_capcity
+				months = 1
```

Comment:
It feels like the monthly_pulse comes at a random time each month, so a country could potentially skip a week or two of this modifier, and also stack it =/